docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,743 @@
+names clash with methods on a model, it would also make it safer to add more methods to a model without risking
+new clashes.
+
+After much deliberation (and even giving a lightning talk at the python language submit about alternatives, see 
+[here](https://discuss.python.org/t/better-fields-access-and-allowing-a-new-character-at-the-start-of-identifiers/14529))
+I've decided to go with the simplest and clearest approach, at the expense of a bit more typing:
+
+All methods on models will start with `model_`, fields' names will not be allowed to start with `"model"`
+(aliases can be used if required).
+
+This will mean the following methods and attributes on a model:
+
+* `.__dict__` as currently, holds a dict of validated data
+* `.__fields_set__` as currently, set containing which fields were set (vs. populated from defaults)
+* `.model_validate()` (currently `.parse_obj()`) - validate data
+* `.model_validate_json()` (currently `parse_raw(j..., content_type='application/json')`) - validate data from JSON
+* `.model_dump()` (currently `.dict()`) - as above, with new `mode` argument
+* `.model_json()` (currently `.json()`) - alias of `.model_dump(mode='json')`
+* `.model_schema()` (currently `.schema()`)
+* `.model_schema_json()` (currently `.schema_json()`)
+* `.model_update_forward_refs()` (currently `.update_forward_refs()`) - update forward references
+* `.model_copy()` (currently `.copy()`) - copy a model
+* `.model_construct()` (currently `.construct()`) - construct a model with no validation
+* `.__model_validator__` attribute holding the internal pydantic `SchemaValidator`
+* `.model_fields` (currently `.__fields__`) - although the format will have to change a lot, might be an alias of
+  `.__model_validator__.schema`
+* `.model_customise_schema()` - a function to customise the schema used to build the `pydantic_core.SchemaValidator`
+  this provides a way to add or change validation without having to use a custom type
+* `ModelConfig` (currently `Config`) - configuration class for models
+
+The following methods will be removed:
+
+* `.parse_file()`
+* `.parse_raw()`
+* `.from_orm()` - the functionality has been moved to config, see [other improvements](#other-improvements) below
+
+### Strict API & API documentation :thumbsup:
+
+When preparing for pydantic V2, we'll make a strict distinction between the public API and private functions & classes.
+Private objects will be clearly identified as private via a `_internal` sub package to discourage use.
+
+The public API will have API documentation. I've recently been working with the wonderful
+[mkdocstrings](https://github.com/mkdocstrings/mkdocstrings) package for both 
+[dirty-equals](https://dirty-equals.helpmanual.io/) and
+[watchfiles](https://watchfiles.helpmanual.io/) documentation. I intend to use `mkdocstrings` to generate complete
+API documentation for V2.
+
+This wouldn't replace the current example-based somewhat informal documentation style but instead will augment it.
```

Comment:
Nice! I've been asked for this recently in FastAPI. :sweat_smile: 

---

Somewhat irrelevant rant no one needs to read:

I would like to find a way to document parameters *inline*, using `Annotated`, instead of having to remember to keep docstrings in sync, also because docstrings are based on some custom format without editor help.

I think some type of parameter docs with `Annotated`, and having a way to annotate/decorate things (functions and classes) and mark them as discouraged/deprecated in a "standard" way would be quite helpful for Python. And if editors supported it, if it was some type of standard, it would help the experience of writing and maintaining code docs.

Reply:
I hadn't thought of `Annotated`, pycharm does a pretty good job of highlighting markdown in docstrings.

On annotating attributes, https://github.com/mkdocstrings/mkdocstrings/issues/446 might be of interest to you.

Have a play with the watchfiles codebase.

Reply:
Nice, thank you!